api/PayrollProcessor.Data.Persistence/Features/Departments/DepartmentEmployeeUpdateCommandHandler.cs

```diff
@@ -20,46 +22,55 @@ public DepartmentEmployeeUpdateCommandHandler(CosmosClient client)
             this.client = client;
         }
 
-        public TryOptionAsync<DepartmentEmployee> Execute(DepartmentEmployeeUpdateCommand command, CancellationToken token)
+        public TryAsync<DepartmentEmployee> Execute(DepartmentEmployeeUpdateCommand command, CancellationToken token)
         {
-            var record = DepartmentEmployeeRecord.Map.Merge(command.Employee, command.DepartmentEmployee);
-
             var departmentsContainer = client.GetDepartmentsContainer();
 
-            return async () =>
-            {
-                var response = await departmentsContainer
+            return DepartmentEmployeeRecord
+                .Map
+                .Merge(command.Employee, command.DepartmentEmployee)
+                .Apply(record => departmentsContainer
                     .ReplaceItemAsync(
                         record, record.Id.ToString(),
                         new PartitionKey(record.PartitionKey),
-                        new ItemRequestOptions { IfMatchEtag = record.ETag }, token);
-
-                var iterator = departmentsContainer
-                    .GetItemLinqQueryable<DepartmentPayrollRecord>(requestOptions: new QueryRequestOptions
-                    {
-                        PartitionKey = new PartitionKey(record.PartitionKey)
-                    })
-                    .Where(p => p.Type == nameof(DepartmentPayrollRecord))
-                    .Where(p => p.EmployeeId == command.Employee.Id)
-                    .ToFeedIterator();
-
-                while (iterator.HasMoreResults)
+                        new ItemRequestOptions { IfMatchEtag = record.ETag }, token)
+                    .Apply(TryAsync)
+                )
+                .Map(CosmosResponse.Unwrap)
+                .SelectMany(record =>
+                    departmentsContainer
+                        .GetItemLinqQueryable<DepartmentPayrollRecord>(requestOptions: new QueryRequestOptions
+                        {
+                            PartitionKey = new PartitionKey(record.PartitionKey)
+                        })
+                        .Where(p => p.Type == nameof(DepartmentPayrollRecord))
+                        .Where(p => p.EmployeeId == command.Employee.Id)
+                        .ToFeedIterator()
+                        .Apply(TryAsync),
+                    (record, iterator) => new { record, iterator }
+                )
+                .MapAsync(async aggregate =>
                 {
-                    foreach (var result in await iterator.ReadNextAsync(token))
+                    var record = aggregate.record;
+                    var iterator = aggregate.iterator;
+
+                    while (iterator.HasMoreResults)
```

Comment:
No clue 🤷‍♂️ what the best techniques are for converting a `foreach` nested in a `while` to use `LanguageExt` types.